client/src/components/DynamicJsonForm.tsx

```diff
@@ -140,36 +140,144 @@ const DynamicJsonForm = ({
+        return schema.required;
+      }
+      if (Array.isArray(schema.required) && fieldPath.length > 0) {
+        return schema.required.includes(fieldPath[fieldPath.length - 1]);
+      }
+      return false;
+    };
+
+    if (propSchema.type === "object" && propSchema.properties) {
+      const objectValue = (currentValue as Record<string, JsonValue>) || {};
+
+      return (
+        <div className="space-y-4">
+          {Object.entries(propSchema.properties).map(
+            ([fieldName, fieldSchema]) => {
+              const fieldPath = [...path, fieldName];
+              const fieldValue = objectValue[fieldName];
+              const fieldRequired = isFieldRequired([fieldName]);
+
+              return (
+                <div key={fieldName} className="space-y-2">
+                  <label
+                    htmlFor={fieldName}
+                    className="block text-sm font-medium"
+                  >
+                    {fieldSchema.title || fieldName}
+                    {fieldRequired && (
+                      <span className="text-red-500 ml-1">*</span>
+                    )}
+                  </label>
+                  {fieldSchema.description && (
+                    <p className="text-xs text-gray-500">
+                      {fieldSchema.description}
+                    </p>
+                  )}
+                  <div>
+                    {renderFieldInput(
+                      fieldSchema,
+                      fieldValue,
+                      fieldPath,
+                      fieldRequired,
+                    )}
+                  </div>
+                </div>
+              );
+            },
+          )}
+        </div>
+      );
+    }
+
+    const fieldRequired = isFieldRequired(path);
+    return renderFieldInput(propSchema, currentValue, path, fieldRequired);
+  };
+
+  const renderFieldInput = (
+    propSchema: JsonSchemaType,
+    currentValue: JsonValue,
+    path: string[],
+    fieldRequired: boolean,
+  ) => {
     switch (propSchema.type) {
-      case "string":
+      case "string": {
+        if (propSchema.enum) {
+          return (
+            <select
+              value={(currentValue as string) ?? ""}
+              onChange={(e) => {
+                const val = e.target.value;
+                if (!val && !fieldRequired) {
+                  handleFieldChange(path, undefined);
+                } else {
+                  handleFieldChange(path, val);
+                }
+              }}
+              required={fieldRequired}
+              className="w-full px-3 py-2 border border-gray-300 rounded-md focus:outline-none focus:ring-2 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-800"
+            >
+              <option value="">Select an option...</option>
+              {propSchema.enum.map((option, index) => (
+                <option key={option} value={option}>
+                  {propSchema.enumNames?.[index] || option}
```

Comment:
`enumNames` [is not part of the JSONSchema spec](https://github.com/rjsf-team/react-jsonschema-form/issues/532). The [recommended way to describe](https://github.com/json-schema-org/json-schema-spec/issues/57) enum labels is with `oneOf` or `anyOf`.